scm/github.go

```diff
@@ -388,26 +388,27 @@ func (s *GithubSCM) createRepository(ctx context.Context, opt *CreateRepositoryO
 }
 
 // deleteRepository deletes repository by name or ID.
-func (s *GithubSCM) deleteRepository(ctx context.Context, opt *RepositoryOptions) error {
+func (s *GithubSCM) deleteRepository(ctx context.Context, id uint64) error {
 	const op Op = "deleteRepository"
 	m := M("failed to delete repository")
-	if !opt.valid() {
-		return E(op, m, fmt.Errorf("missing fields: %+v", *opt))
+	if id == 0 {
+		return E(op, m, fmt.Errorf("missing ID"))
 	}
 
 	// if ID provided, get path and owner from github
-	if opt.ID > 0 {
-		repo, _, err := s.client.Repositories.GetByID(ctx, int64(opt.ID))
+	if id > 0 {
+		repo, _, err := s.client.Repositories.GetByID(ctx, int64(id))
 		if err != nil {
-			return E(op, m, fmt.Errorf("failed to get repository %d: %w", opt.ID, err))
+			return E(op, m, fmt.Errorf("failed to get repository %d: %w", id, err))
 		}
-		opt.Repo = repo.GetName()
-		opt.Owner = repo.Owner.GetLogin()
-	}
+		var repo_name = repo.GetName()
+		var repo_owner = repo.Owner.GetLogin()
```

Comment:
We should avoid using underscores in names and use [`MixedCaps`](https://go.dev/doc/effective_go#mixed-caps) instead:

```go
// would be
repoName := repo.GetName()
// alternative
name := repo.GetName()
```

For variables that have explicit values we tend to use shorthand declarations, e.g. `owner := repo.Owner.GetLogin()`

We could also skip this step altogether and use the variables directly in `.Delete(...)`

```go
if _, err := s.client.Repositories.Delete(ctx, repo.GetName(), repo.Owner.GetLogin())
```

Reply:
Yes. The last alternative without using variables is my preference here; there is almost no cost to calling the Getter-methods within the `Delete` call, and calling them again if an error occurs should not happen often.

However, to be sure that that this code doesn't cause a panic, due to a nil pointer exception, we should use `repo.GetOwner().GetLogin()`:
```go
if _, err := s.client.Repositories.Delete(ctx, repo.GetName(), repo.GetOwner().GetLogin())
```
Using `GetOwner` and `GetLogin` is safe against calls on a `nil` object.